### PR TITLE
Add the examples and the tests in the installed packages. 

### PR DESCRIPTION
**Pull request by nouiz**
This way, we don't need to keep the source directory to run all the tests and examples. This is small.

### CODE REVIEW

**Comment by yhtang** [comment]
Note that one issue with this proposed approach is that the `examples` and `tests` folders will be directly placed in the Python site-packages folder alongside the `transformer_engine` engine. As such, it will not be obvious which software these folders belong to.

Maybe, they should be placed inside the transformer_engine folder?

**Comment by denera** [comment]
@nouiz The folder hierarchy should be correct now with the latest commit, but this involves turning examples into an importable submodule. In order to make that a bit cleaner, I migrated the example-based JAX unit tests into the tests folder.

Unittests are copied over as "package data" so they're not importable. I didn't think it made much sense for them to be submodules.

**Review by nouiz — COMMENTED** [review]
Looks good overall.
@mingxu1067 can you also check?
@yhtang @DwarKapex, can you confirm this will fix the original issue?

File: setup.py [hunk 1]
```diff
@@ -608,10 +641,11 @@ def main():
         ext_modules.append(setup_paddle_extension())
 
     # Configure package
-    setuptools.setup(
+    s = setuptools.setup(
```
[the remark this answers is not in the document]
Comment (denera) [reply setup.py:644]: This is unintentionally leftover from debugging. I will remove it shortly alongside other minor fixes.

**Comment by ksivaman** [comment]
This PR is stale as it was created before the build system overhaul and as a result has several conflicts. After a discussion with @nouiz, we will re-open a fresh PR.

**Pull request closed without merging**